test/e2e-tests.js

```diff
@@ -106,6 +107,23 @@ describe('matrix-public-archive', () => {
   });
 
   describe('Archive', () => {
+    before(async () => {
+      // FIXME: This doesn't work because `400 Bad Request:
+      // {"errcode":"M_EXCLUSIVE","error":"This user ID is reserved by an
+      // application service."}` and we can't easily remove the AS registration,
+      // register the user, add the AS back, restart Synapse, and continue the
+      // tests.
+      //
+      // Make sure the application service archiver user itself is also
+      // explicitly registered otherwise we run into 404, `Profile was not
+      // found` errors when joining a remote federated room from the archiver
+      // user, see https://github.com/matrix-org/synapse/issues/4778
```

Comment:
This doesn't work yet, thanks to https://github.com/matrix-org/synapse/issues/4778

And we can't register the username of the application service defined in the registration files.